### PR TITLE
cmake: Improve Boost checks

### PR DESCRIPTION
**Pull request by hebasto**
The first commit was requested offline.

The second commit speeds up configuration.

### CODE REVIEW

**Comment by hebasto** [comment]
> The first commit was requested offline.

It addresses @fanquake's concerns:
> Why does AddBoostIfNeeded check for two different boost test headers ? We don't do this in master.

> Ok. So it's another Windows package manager specific work around, being applied to all platforms. It'd be good to document that.

File: cmake/module/AddBoostIfNeeded.cmake [hunk 1]
```diff
@@ -55,22 +55,16 @@ function(add_boost_if_needed)
...
       message(FATAL_ERROR "Building test_bitcoin executable requested but boost/test/included/unit_test.hpp header not available.")
     endif()
-
-    check_cxx_source_compiles("
-      #define BOOST_TEST_MAIN
-      #include <boost/test/included/unit_test.hpp>
-      #include <boost/test/unit_test.hpp>
-      " HAVE_BOOST_UNIT_TEST_H
-    )
+    check_include_file_cxx(boost/test/unit_test.hpp HAVE_BOOST_UNIT_TEST_H)
```
Comment (fanquake) [inline cmake/module/AddBoostIfNeeded.cmake:67]: I think I might be missing something, but this is still checking two headers.

Comment (hebasto) [reply cmake/module/AddBoostIfNeeded.cmake]: Thanks! Reworked.

Only one Boost.Test header is being checked now.

File: cmake/module/AddBoostIfNeeded.cmake [hunk 2]
```diff
@@ -55,25 +55,15 @@ function(add_boost_if_needed)
   endif()
 
   if(BUILD_TESTS)
-    include(CheckCXXSourceCompiles)
-    check_cxx_source_compiles("
-      #define BOOST_TEST_MAIN
-      #include <boost/test/included/unit_test.hpp>
-      " HAVE_BOOST_INCLUDED_UNIT_TEST_H
-    )
+    # Some package managers, such as vcpkg, install Boost headers individually
```
Comment (fanquake) [inline cmake/module/AddBoostIfNeeded.cmake:58]:
```suggestion
    # Some package managers, such as vcpkg, vendor Boost Test separately
    # from the rest of the headers, so we have to check for it individually.
```

**Review by fanquake — COMMENTED** [review]
Looks good, just think that comment could be slightly clearer

File: cmake/module/AddBoostIfNeeded.cmake (hunk 2 above)
Comment (hebasto) [reply cmake/module/AddBoostIfNeeded.cmake]: Thanks! Done.

**Review by fanquake — APPROVED** [review]

**Pull request merged**